### ISSUE TITLE
Meta-issue: Relicense under dual MIT / Apache v2

### ISSUE THREAD

**Issue by Manishearth** [rust-lang-nursery/rust-clippy#3093]
Previously: https://github.com/rust-lang-nursery/rust-clippy/issues/2885
The majority of the Rust ecosystem, and the Rust compiler itself, uses the dual MIT/Apache-2.0 license. This means that people can choose which of these two licenses to use a library under.

As Clippy starts [uplifting lints to the compiler](https://github.com/rust-lang/rust/issues/53224) and generally being more accessible, I'd like to switch to this license for consistency. We currently use the copyleft MPL-2.0 license. This license is only file-level infectious, but still may hamper efforts to uplift lint code to the compiler. This initial choice of license was pretty random, and at the time the Rust ecosystem didn't have a consistent license.

For reasons why the Rust ecosystem uses this particular dual license, see [the explainer text used when the community switched over](https://github.com/cmr/relicense-assistant/blob/master/issue-template.txt#L14)

This requires each one of our contributors to sign off, and I'll be opening sub-issues for this.

**Comment by Manishearth** [comment]
See #3094, #3095, #3096, #3097, #3098, #3099, #3100 for the sub-issues. I split them up to avoid notification spam, folks can mute the issues if they wish.

**Comment by mati865** [comment]
It's been over two weeks, guess you could ping missing folks again or close those issues and create new one with those who didn't respond yet and new contributors.

**Comment by Manishearth** [comment]
Yeah, I plan to do something like that, but I'm travelling a bit.

Feel free to do it if you'd like! Try to include new contributors too.

(I was also considering sending out emails, though that might be very nag-y)

**Comment by Manishearth** [comment]
#3095 is full! six more to go :smile: 

I'm closing issues as they get filled out.

**Comment by Manishearth** [comment]
And #3100 is done

**Comment by mati865** [comment]
Pinging was helpful a bit but there are still contributors who didn't sign and mailing them seems to be the only way to bring their attention.

Maybe it's not worth to chase every contributor and just revert the commit if changes are small?
See [this example](https://github.com/rust-lang-nursery/rust-clippy/commits?author=zmt00), modified README.md section was already removed so revert would just bring back typo.
I'm not an licensing expert but there is just one way to fix typo so we could just fix it afterwards.

**Comment by flip1995** [comment]
List of PRs (only 1 PR per contributor) with contributors not appearing on one of the lists, that were merged after this issue was opened:
- [ ] #3076
- [ ] #3105
- [ ] #3108
- [ ] #3131
- [ ] #3152
- [ ] #3169 (only typo fixes)
- [ ] #3191
- [x] #3195 (already accepted the licence change)

I will add PRs of new contributors to this list, so we could ping them after we're done with the old contributors.

**Comment by Manishearth** [comment]
#3094 is done. Three more to go.

**Comment by Manishearth** [comment]
Opened https://github.com/rust-lang-nursery/rust-clippy/issues/3230 for the new contributors

**Comment by Manishearth** [comment]
New contributors done.

We just have three people left, @VKlayd , @wartman4404 , and @EpocSquadron 

I've emailed all of them, however @wartman4404's email address bounces. We may need to rip out and redo their work. Anyone want to work with me on that? Ideally you have two people do it -- one person removes the code and describes what it did to the other, who reimplements it.

**Comment by flip1995** [comment]
One new contributor is missing: #3178 was done by @ms2300, not `@frewsxcv`, I already ticked the PR of `@frewsxcv` this morning.

**Comment by Manishearth** [comment]
Okay, so aside from #3178 we have two missing contributors. I removed their changes in https://github.com/rust-lang-nursery/rust-clippy/pull/3251 , I need folks who have not seen their code to rewrite it and make a PR to that branch. I've included instructions there for what needs to be written so that you don't have to look at the actual code. Any volunteers? :smile: 

I'll also double-check that no new contributors have showed up since then.

**Comment by Manishearth** [comment]
Okay, so, we have not gotten sign-off from:

 - joshtriplett: contributed after this issue was opened, forgot to ask, asked now. It's minor and ignorable
 - EpocSquadron: obsolete readme contributions that have been overwritten
 - ms2300: contributed after the issue was opened, hasn't responded yet. It's a newish PR which we can revert if we have to
 - VKlayd: Nonminor contributions, being rewritten in #3251 
 - wartman4404: ditto
 - zmt00: minor typo fixes, ignorable

So once we finish #3251 and ensure and get sign-off from ms2300, we should be all set!

**Comment by Manishearth** [comment]
I'm also going to check in with the legal team at Mozilla on Friday to ensure we've done everything right. Once we finish that (and the last piece of #3251 goes in), we can relicense!

**Comment by Manishearth** [comment]
#3251 merged. I'm going to do a final check to ensure nobody new has snuck in since we ran our checks, and then complete this!

**Comment by Manishearth** [comment]
So the Github API seems buggy -- the user AVerm, who previously showed up on my API fetches on `https://api.github.com/repos/rust-lang-nursery/rust-clippy/contributors?per_page=100&page=...` no longer does, [despite them having contributed](https://github.com/rust-lang-nursery/rust-clippy/commits?author=AVerm).

So I ran a similar script on `https://api.github.com/repos/rust-lang-nursery/rust-clippy/commits?per_page=...` and it turned up https://github.com/rust-lang-nursery/rust-clippy/commits?author=sanmai-NL , who has made a very minor contribution and can be ignored.

**Comment by wartmanm** [comment]
I am so sorry about this!  I never updated my email because I figured no one would need to contact me, and then I compounded my mistake by not logging in for a year.

**Comment by Manishearth** [comment]
@wartmanm no problem! It still would be useful if you posted a sign off in one of the sub-issues ; even if we rewrote the code it's good to have the sign off
